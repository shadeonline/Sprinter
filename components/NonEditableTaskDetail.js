import React, { useState, useEffect } from 'react';
import { Modal, StyleSheet, Text, TouchableOpacity, View, TextInput } from 'react-native';
import { useNavigation } from '@react-navigation/core';
import Icon from 'react-native-vector-icons/Ionicons'; // You can choose any icon library you prefer
import SelectDropdown from 'react-native-select-dropdown'
import { firebaseEditTask } from "../firebaseFunctions";

const NonEditableTaskDetail = ({ task, setIsEditing, handleDeleteTask }) => {
    const navigation = useNavigation();

    const [editedTask, setEditedTask] = useState(task);


    const handleStatusChange = (selectedStatus) => {
        // Update the editedTask state with the new status
        setEditedTask((prevEditedTask) => ({
            ...prevEditedTask,
            status: selectedStatus,
        }));
    }

    useEffect(() => {
        // This effect will run whenever editedTask changes
        // Check if the status has changed and then update it in Firebase
        if (editedTask.status !== task.status) {
            // Call the function to edit the task in Firebase
            firebaseEditTask(editedTask)
                .then((success) => {
                    if (success) {
                        console.log("Task edited successfully");
                    } else {
                        console.log("Error editing task");
                    }
                })
                .catch((error) => {
                    console.error("Error editing task:", error);
                });
        }
    }, [editedTask]);


    // console.log(task.status);
    return (
        <View style={styles.modalDetails}>

            {/* Add icons for Edit and Delete buttons */}
            <View style={styles.iconContainer}>
                <TouchableOpacity onPress={() => setIsEditing(true)}>
                    <Icon name="pencil-outline" size={24} color="#0052CC" style={styles.icon} />
                </TouchableOpacity>
                <TouchableOpacity onPress={handleDeleteTask}>
                    <Icon name="trash-outline" size={24} color="#DD2C00" style={styles.icon} />
                </TouchableOpacity>
            </View>
            <Text style={styles.modalLabel}>Status:</Text>


            <SelectDropdown
                defaultValue={task.status}
                data={["-", "In Progress", "Completed"]}
                onSelect={(selectedStatus) => {
                    handleStatusChange(selectedStatus)
                }}
                buttonStyle = {styles.selectButton}
                buttonTextStyle ={styles.selectButtonText}
                
            />
            <Text style={styles.modalLabel}>Storypoint:</Text>
            <Text style={styles.modalText}>{task.storyPoint}</Text>
            <Text style={styles.modalLabel}>Description:</Text>
            <Text style={styles.modalText}>{task.taskDescription}</Text>
            <Text style={styles.modalLabel}>Deadline:</Text>
            <Text style={styles.modalText}>{task.deadline}</Text>

            <TouchableOpacity
                style={styles.button}
                onPress={() => navigation.goBack()}
            >
                <Text style={styles.buttonText}>Back</Text>
            </TouchableOpacity>
        </View>
    );
};

const styles = StyleSheet.create({
    modalDetails: {
        padding: 16,
        backgroundColor: 'white',
    },
    modalTitle: {
        fontSize: 20,
        fontWeight: 'bold',
        marginBottom: 16,
    },
    modalLabel: {
        fontSize: 16,
        fontWeight: 'bold',
        marginBottom: 8,
        color: '#949396', // Jira-like label color
    },
    modalText: {
        fontSize: 16,
        color: 'black',
        marginBottom: 16,
    },
    iconContainer: {
        flexDirection: 'row',
        justifyContent: 'flex-end', // Align icons to the right
        marginBottom: 16,
    },
    icon: {
        marginHorizontal: 8,
    },
    button: {
        padding: 12,
        borderRadius: 4,
        alignItems: 'center',
        marginTop: 10,
        backgroundColor: 'grey'
    },
    buttonText: {
        color: 'white',
        fontWeight: 'bold',
    },

    selectButton: {
        padding: 12,
        alignItems: 'center',
        backgroundColor: '#5386e4',
        marginBottom: 16,
        width: '40%',
        height: '10%'
    },
    selectButtonText: {
        color: 'white',
        fontWeight: 'bold',
        fontSize: 13,
    },

    statusDropdownContainer: {
        marginBottom: 16,
        borderColor: '#949396',
        borderWidth: 1,
        borderRadius: 4,
    },
    statusDropdown: {
        height: 40,
        width: '100%',
    },
});

export default NonEditableTaskDetail;
