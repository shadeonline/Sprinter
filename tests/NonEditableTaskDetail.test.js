import React from 'react';
import renderer from 'react-test-renderer';
import NonEditableTaskDetail from '../components/NonEditableTaskDetail';
import { NavigationContainer } from '@react-navigation/native';


// Mock firebaseEditTask function
jest.mock('../firebaseFunctions', () => ({
  firebaseEditTask: jest.fn(),
}));


// Mock StackNavigator
jest.mock('@react-navigation/stack', () => {
  const actualStack = jest.requireActual('@react-navigation/stack');
  return {
    ...actualStack,
    createStackNavigator: () => ({
      Navigator: ({ children }) => <>{children}</>,
      Screen: ({ children }) => <>{children}</>,
    }),
  };
});

describe('NonEditableTaskDetail', () => {
  it('matches snapshot', () => {
    const task = {
      taskTitle: 'Sample Task',
      storyPoint: '3',
      taskDescription: 'This is a sample task description.',
      deadline: '30/09/23',
      status: 'Backlog',
    };
    const setIsEditing = jest.fn();
    const handleDeleteTask = jest.fn();

    const tree = renderer
      .create(
        <NavigationContainer>
          <NonEditableTaskDetail
            task={task}
            setIsEditing={setIsEditing}
            handleDeleteTask={handleDeleteTask}
          />
        </NavigationContainer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
