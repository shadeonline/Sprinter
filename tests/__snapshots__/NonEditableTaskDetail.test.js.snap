// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonEditableTaskDetail matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginBottom": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <Text
    style={
      {
        "color": "#949396",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Status:
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#999999",
        "flexDirection": "row-reverse",
        "height": "10%",
        "justifyContent": "space-between",
        "marginBottom": 16,
        "opacity": 1,
        "overflow": "hidden",
        "padding": 12,
        "paddingHorizontal": 8,
        "width": "40%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      numberOfLines={1}
      style={
        {
          "color": "white",
          "flex": 1,
          "fontSize": 13,
          "fontWeight": "bold",
          "marginHorizontal": 8,
          "textAlign": "center",
        }
      }
    >
      Select an option.
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#949396",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Storypoint:
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 16,
        "marginBottom": 16,
      }
    }
  >
    3
  </Text>
  <Text
    style={
      {
        "color": "#949396",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Description:
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 16,
        "marginBottom": 16,
      }
    }
  >
    This is a sample task description.
  </Text>
  <Text
    style={
      {
        "color": "#949396",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Deadline:
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 16,
        "marginBottom": 16,
      }
    }
  >
    30/09/23
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 4,
        "marginTop": 10,
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Back
    </Text>
  </View>
</View>
`;
